src/trackteroid/entities/base.py

```diff
@@ -425,22 +447,50 @@ def __getattr__(self, item):
                 else:
                     values.append(value)
 
+            # If all values are NOT_SET means we have not fetched that attribute
+            # that we're trying to retrieve. We must then pick up reference and
+            # collection attributes and properly return them as EmptyCollection
+            # of the proper entity type.
+            if values and all([x is NOT_SET for x in values]):
+                # We get the ftrack attribute type, to check if this attribute is a reference
+                # to any other type of entity
+                ftrack_attribute = sample_entity.attributes.get(item)
+
+                if isinstance(ftrack_attribute, (CollectionAttribute, ReferenceAttribute)):
+
+                    # If it's a collection or a reference attribute, we'll replace NOT_SET with
+                    # an empty collection of the right type, for that we need to determine what
+                    # type of attribute it is via its original Ftrack type schema
+                    ftrack_session = sample_entity.session
```

Comment:
Is there a particular advantage of referring to the original Ftrack session like this rather than the collection's session instance we can retrieve via `self._session`? We are referring to `self._session` down the line so I'd like to have it consistent if possible.